spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/ProxyRequestHelper.java

```diff
@@ -156,23 +156,16 @@ public void setResponse(int status, InputStream entity,
 			context.setResponseDataStream(entity);
 		}
 
-		HttpHeaders httpHeaders = new HttpHeaders();
```

Comment:
Actually, a benchmark whether this actually improves performance or not would be useful. My gut tells me calling `equalsIgnoreCase` is probably slower than the built-in case insensitivity in Spring's `HttpHeaders`.